tests/structured_conf/test_structured_config.py

```diff
@@ -1010,6 +1010,26 @@ def test_str2user(self, module: Any) -> None:
             # bad key
             cfg[Color.BLUE] = "nope"
 
+    def test_str2user_with_field(self, module: Any) -> None:
+        cfg = OmegaConf.structured(module.DictSubclass.Str2UserWithField())
+
+        assert cfg.foo.name == "Bond"
+        assert cfg.foo.age == 7
+        assert isinstance(cfg.foo, DictConfig)
+
+        cfg.mp = module.User(name="Moneypenny", age=11)
+        assert cfg.mp.name == "Moneypenny"
+        assert cfg.mp.age == 11
+        assert isinstance(cfg.mp, DictConfig)
+
+        with raises(ValidationError):
+            # bad value
+            cfg.hello = "world"
+
+        with raises(KeyValidationError):
+            # bad key
+            cfg[Color.BLUE] = "nope"
```

Comment:
Can you assign a valid user here instead of a string? (ex: cfg.mp)

Reply:
Done in https://github.com/omry/omegaconf/pull/502/commits/7f8addbb7ca846edd672060c6d64fae494d167f6.